### PR TITLE
Sam non-SA Service Account test

### PR DESCRIPTION
**Pull request by jmthibault79**
- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [ ] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by jmthibault79** [comment]
Should be rebased after #1048 is merged

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/api/sam/SamApiSpec.scala [hunk 1]
```diff
@@ -63,6 +67,50 @@ class SamApiSpec extends FreeSpec with Matchers {
...
+        programLocationState = "MA",
+        programLocationCountry = "USA",
+        pi = "Albus Dumbledore",
+        nonProfitStatus = "true"
+      )
+      Orchestration.profile.registerUser(newUserProfile)
+    }
+
+    "should not treat non-pet service accounts as pets" in {
+      val saEmail = WorkbenchUserServiceAccountEmail(Config.GCS.qaEmail)
```
Comment (rtitle) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/api/sam/SamApiSpec.scala:96]: Cool, using the QA service account. 👍

**Review by rtitle — APPROVED** [review]
It looks good to me

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/api/sam/SamApiSpec.scala [hunk 2]
```diff
@@ -24,9 +24,13 @@ class SamApiSpec extends FreeSpec with Matchers {
     Await.result(remove, 5.seconds)
   }
 
+  def findSaInGoogle(name: WorkbenchUserServiceAccountName): Option[WorkbenchUserServiceAccount] = {
+    val find = googleIamDAO.findServiceAccount(GoogleProject(Config.Projects.default), name)
+    Await.result(find, 1.minute)
```
Comment (ansingh7115) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/api/sam/SamApiSpec.scala]: Is this wait time arbitrary? Do the tests consistently pass with this wait time?

Comment (jmthibault79) [reply automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/api/sam/SamApiSpec.scala]: it is arbitrary - and way too long.  Let me reduce it.

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/api/sam/SamApiSpec.scala [hunk 3]
```diff
@@ -63,6 +67,50 @@ class SamApiSpec extends FreeSpec with Matchers {
...
+      )
+      Orchestration.profile.registerUser(newUserProfile)
+    }
+
+    "should not treat non-pet service accounts as pets" in {
+      val saEmail = WorkbenchUserServiceAccountEmail(Config.GCS.qaEmail)
+      val sa = findSaInGoogle(saEmail.toAccountName).get
+
+      // ensure clean state: SA's user not registered
+      cleanupUser(sa.subjectId)
```
Comment (ansingh7115) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/api/sam/SamApiSpec.scala:100]: Question - will "cleaning up" the QA service account affect anything else? Just want to make sure it's not being used for other things

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/api/sam/SamApiSpec.scala [hunk 4]
```diff
@@ -63,6 +67,50 @@ class SamApiSpec extends FreeSpec with Matchers {
...
+
+      implicit val saAuthToken: ServiceAccountAuthToken = ServiceAccountAuthToken(saEmail)
+
+      registerSaAsNewUser(saEmail)
+
+      // I am no one's pet.  I am myself.
```
Comment (ansingh7115) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/api/sam/SamApiSpec.scala:106]: Nobody puts my service account in a corner!

**Review by ansingh7115 — APPROVED** [review]
Seems great!

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/api/sam/SamApiSpec.scala (hunk 3 above)
Comment (rtitle) [reply automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/api/sam/SamApiSpec.scala:100]: I wondered about that -- but "clean up" seems to just clean up in Sam LDAP, NOT delete the SA in Google (which would be bad). @jmthibault79 confirm?

Comment (jmthibault79) [reply automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/api/sam/SamApiSpec.scala:100]: This removes the SA's *associated Firecloud user* which did not exist before I made this test

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/api/sam/SamApiSpec.scala [hunk 5]
```diff
@@ -24,11 +24,15 @@ class SamApiSpec extends FreeSpec with Matchers {
     Await.result(remove, 5.seconds)
   }
 
-  def findPetInGoogle(userInfo: UserStatusDetails): Option[WorkbenchUserServiceAccount] = {
-    val find = googleIamDAO.findServiceAccount(GoogleProject(Config.Projects.default), petName(userInfo))
+  def findSaInGoogle(name: WorkbenchUserServiceAccountName): Option[WorkbenchUserServiceAccount] = {
+    val find = googleIamDAO.findServiceAccount(GoogleProject(Config.Projects.default), name)
     Await.result(find, 5.seconds)
```
Comment (ansingh7115) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/api/sam/SamApiSpec.scala]: Why are we cutting it off at 5 seconds? Are there times the Future just hangs?

Comment (jmthibault79) [reply automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/api/sam/SamApiSpec.scala]: shrug-emoji we have to pick some value

Comment (jmthibault79) [reply automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/api/sam/SamApiSpec.scala]: and 1 sec is too short

**Pull request merged**